testing/nessie-container/src/main/java/org/projectnessie/testing/nessie/NessieContainer.java

```diff
@@ -94,9 +102,24 @@ public interface Builder {
       @CanIgnoreReturnValue
       Builder authEnabled(boolean authEnabled);
 
+      default Builder oidcFromCustomKeycloakContainer(
+          CustomKeycloakContainer customKeycloakContainer) {
```

Comment:
If I'm not mistaken, this should work even if callers do not have the `:nessie-keycloak-testcontainer` dependency.

Reply:
I tried to create a project using the Nessie container without the Keycloak container, but the test failed with:

```
Caused by: java.lang.NoClassDefFoundError: org/projectnessie/testing/keycloak/CustomKeycloakContainer
	at org.projectnessie.testing.nessie.NessieContainer$NessieConfig$Builder.oidcFromCustomKeycloakContainer(NessieContainer.java:107)
	at org.projectnessie.quarkus.tests.profiles.NessieDockerTestResourceLifecycleManager.start(NessieDockerTestResourceLifecycleManager.java:77)
	at io.quarkus.test.common.TestResourceManager$TestResourceEntryRunnable.run(TestResourceManager.java:481)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
	... 3 more
Caused by: java.lang.ClassNotFoundException: org.projectnessie.testing.keycloak.CustomKeycloakContainer
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 7 more
```

That said, the test was using the Nessie Quarkus test resource, which may have a stronger dependency to Keycloak than the bare container. Although the error is being thrown in `org.projectnessie.testing.nessie.NessieContainer$NessieConfig$Builder.oidcFromCustomKeycloakContainer` so 🤷‍♂️ 

How about inverting the dependency graph? I.e., let Keycloak container depend on Nessie container and move `oidcFromCustomKeycloakContainer` to the Keycloak module? (I did not test it).

Reply:
Damn - I thought, it would require the class once the method's used. But yea, my bad - that only works if a class is "first used" in a method body.